src/main/java/org/eclipse/tractusx/ssi/lib/crypt/x21559/x21559Generator.java

```diff
@@ -51,13 +51,13 @@ public KeyPair generateKey() throws KeyGenerationException {
     x21559PrivateKey x21559PrivateKey;
     try {
       x21559PrivateKey = new x21559PrivateKey(privateKey.getEncoded());
-    } catch (InvalidePrivateKeyFormat e) {
+    } catch (InvalidPrivateKeyFormatException e) {
       throw new KeyGenerationException(e.getCause());
```

Comment:
It would be better if InvalidPrivateKeyFormatException extend KeyGenerationException. This would eliminate the need to wrap the message in a KeyGenerationException and rethrow it.